src/fast-instantiator.ts

```diff
@@ -100,6 +102,7 @@ class InstantiatorBuilder<T extends IClassModelType<Record<string, IAnyType>, an
           for (const resolver of context.referencesToResolve) {
             resolver();
           }
+          delete context.referencesToResolve; // cleanup these closures, no need to retain them past construction
```

Comment:
🚀

Reply:
Also, not sure which would be faster: deleting and changing the shape, or setting it to `null` / `undefined`?

Reply:
I tried it, the delete was quite a bit faster locally for me 🤷

Reply:
nvm, delete was actually slower but the benches were obscured by other changes in this PR. Split this bit out into its own PR so we can see the performance impact independently.